LiteFarmOrg/LiteFarm
# Draft: Feature/ensemble communications: bulk upload sensor flow

This PR completes the following tickets: [LF-2483](https://lite-farm.atlassian.net/browse/LF-2483), [LF-2453](https://lite-farm.atlassian.net/browse/LF-2453), [LF-2482](https://lite-farm.atlassian.net/browse/LF-2482), [LF-2451](https://lite-farm.atlassian.net/browse/LF-2451), [LF-2456](https://lite-farm.atlassian.net/browse/LF-2456), [LF-2452](https://lite-farm.atlassian.net/browse/LF-2452), [LF-2460](https://lite-farm.atlassian.net/browse/LF-2460)

This PR has parts of the following tickets which are still in progress:  LF-2454, LF-2484, LF-2458, LF-2459

To test:
- Review [the bulk upload flow diagram](https://lucid.app/lucidchart/57b61ad1-0fa5-4fb4-bdb8-e753c2cd549f/edit?viewport_loc=-2090%2C-689%2C4992%2C2514%2C0_0&invitationId=inv_b24cabbe-67d9-457d-853e-30e351438487#) to remind yourself of the flow.
- Go to the map -> Click "+" -> Scroll down to sensor
- The bulk upload modal should pop up
- Use this csv to test  
[Stamp farm sensors.csv](https://github.com/LiteFarmOrg/LiteFarm/files/8928649/Stamp.farm.sensors.csv)
(you probably want to use only 4 rows for testing purposes, since you will need to manually unclaim to retest)
- If you need to unclaim, POST `https://api.esci.io/organizations/{organization_id}/devices/unclaim/` with body `{"esid": INSERT_EXTERNAL_ID}`
